mockOSApi.v2/Services/ProcessHandler.cs

```diff
@@ -1,66 +1,82 @@
 using System.Text.Json;
 using mockOSApi.Repository;
 using mockOSApi.Models;
+using mockOSApi.DTO;
 
 namespace mockOSApi.Services;
 
 
-public interface IProcessHandler {
-    public IEnumerable<MockProcess> AllProcesses { get; }
+public interface IProcessHandler
+{
+    public IEnumerable<ProcessDto> AllProcesses { get; }
 
-    public MockProcess? GetProcessByPid(int pid);
-      public Task CreateProcess(MockProcess process);
-      public Task KillProcess(MockProcess proc);
-      public void ChangePriority(int prio,int pid);
+    public ProcessDto GetProcessByPid(int pid);
+    public Task CreateProcess(ProcessCreationDto proc);
+    public void KillProcess(ProcessDto proc);
+    public void ChangePriority(int prio, int pid);
 }
 
-public class ProcessHandler: IProcessHandler {
+public class ProcessHandler : IProcessHandler
+{
 
     public readonly IProcessRepository _repository;
+    public readonly IMapper _mapper;
 
-    public ProcessHandler(IProcessRepository procRepo) {
-         _repository = procRepo;
-
+    public ProcessHandler(IProcessRepository procRepo, IMapper mapper)
+    {
+        _repository = procRepo;
+        _mapper = mapper;
     }
 
-    public IEnumerable<MockProcess>? AllProcesses
+    public IEnumerable<ProcessDto>? AllProcesses
     {
 #pragma warning disable CS8766 
 
         get
 #pragma warning restore CS8766 
 
         {
-
             if (_repository.GetAll().ToList().Count == 0)
             {
                 return null;
             }
-            return _repository.GetAll().ToList();
+            List<ProcessDto> DtoList = new List<ProcessDto>();
+            foreach (var proc in _repository.GetAll().ToList())
+            {
+                DtoList.Add(_mapper.MapToDTO(proc));
+            }
+            return DtoList;
+
         }
     }
 
-    public MockProcess? GetProcessByPid(int pid) => _repository.GetProcessByPid(pid);
+    public ProcessDto GetProcessByPid(int pid) => _mapper.MapToDTO(_repository.GetProcessByPid(pid));
+
+    public async Task CreateProcess(ProcessCreationDto process)
+    {
 
-    public async Task CreateProcess(MockProcess process) {
-        await _repository.CreateProcess(process);
+        await _repository.CreateProcess(_mapper.MapFromDTO(process));
         await _repository.Save();
     }
 
-    public async Task KillProcess(MockProcess proc) {
-        _repository.KillProcess(proc);
-        await _repository.Save();
+    public void KillProcess(ProcessDto proc)
+    {
+        _repository.KillProcess(_mapper.MapFromDTO(proc));
+
     }
 
-    public void ChangePriority(int prio,int pid) {
-        if (prio == 50) { // default priority
+    public void ChangePriority(int prio, int pid)
+    {
+        if (prio == 50)
```

Comment:
Use an Enum for this

Reply:
added a const for DEFAULT_PRIORITY since in rest of code the prio can take arbitrary int values